File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> num_list) {
    int e1 = num_list[num_list.size() - 2];
    int e2 = num_list[num_list.size() - 1];
    num_list.push_back(e1 < e2 ? e2 - e1 : e2 * 2);
    return num_list;
}

File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.java
class Solution {
    public int[] solution(int[] num_list) {
        int[] ans = new int[num_list.length + 1];
        for(int i = 0; i < num_list.length; ++i) ans[i] = num_list[i];
        
        int e1 = num_list[num_list.length - 2];
        int e2 = num_list[num_list.length - 1];
        if(e1 < e2){
            ans[ans.length - 1] = e2 - e1;
        } else{
            ans[ans.length - 1] = e2 * 2;
        }
        return ans;
    }
}